docs/user-guide/KubernetesCurator.rst

```diff
@@ -0,0 +1,382 @@
+your StorageClass.
+
+This example requests ``150Gi`` of space. Adjust that number for your
+workloads and be aware that not all storage provisioners support volume
+resizing.
+
+.. code-block:: bash
+
+    STORAGE_CLASS=<...>
+    PVC_NAME=nemo-workspace
+
+    kubectl apply -f - <<EOF
+    apiVersion: v1
+    kind: PersistentVolumeClaim
+    metadata:
+      name: ${PVC_NAME}
+    spec:
+      accessModes:
+        - ReadWriteMany
+      storageClassName: ${STORAGE_CLASS}
+      resources:
+        requests:
+          # Requesting enough storage for a few experiments
+          storage: 150Gi
+    EOF
+
+.. note::
+    The storage class must support ``ReadWriteMany`` because multiple Pods
+    may need to access the PVC to concurrently read and write.
+
+Setup PVC Busybox Helper Pod
+############################
+
+Inspecting the PVC and copying to and from it is facilitated with a busybox container.
+Some examples below assume you have this Pod running to copy to and from the PVC. 
+
+.. code-block:: bash
+
+    PVC_NAME=nemo-workspace
+    MOUNT_PATH=/nemo-workspace
+
+    kubectl create -f - <<EOF
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      name: nemo-workspace-busybox
+    spec:
+      containers:
+      - name: busybox
+        image: busybox
+        command: ["sleep", "infinity"]
+        volumeMounts:
+        - name: workspace
+          mountPath: ${MOUNT_PATH}
+      volumes:
+      - name: workspace
+        persistentVolumeClaim:
+          claimName: ${PVC_NAME}
+    EOF
+
+Feel free to delete this container if no longer needed, but it should use very little
+resources when idle.
+
+.. code-block:: bash
+
+    kubectl delete pod nemo-workspace-busybox
+
+Setup Docker Secrets
+--------------------
+A Kubernetes Secret needs to be created on the k8s cluster to authenticate with the NGC
+private registry. If not done already, get an NGC key from ngc.nvidia.com. Create a
+secret key on the k8s cluster with (replace ``<NGC KEY HERE>`` with your NGC secret
+key. Note that if you have any special characters in your key you might need to wrap
+the key in single quotes (``'``) so it can be parsed correctly by k8s)::
+
+    kubectl create secret docker-registry ngc-registry --docker-server=nvcr.io --docker-username=\$oauthtoken --docker-password=<NGC KEY HERE>
+
+Setup Python Environment
+------------------------
+
+Setup a virtual environment:
+
+.. code-block:: bash
+
+    python3 -m venv venv
+    source venv/bin/activate
+
+    pip install 'dask_kubernetes>=2024.4.1'
+
+Upload Data to PVC
+------------------
+
+To copy into the ``nemo-workspace`` PVC, we will do so with ``kubectl exec``. You may also
+use ``kubectl cp``, but ``exec`` has fewer surprises regarding compressed files:
+
+.. code-block:: bash
+  
+    # Replace <...> with a path on your local machine 
+    LOCAL_WORKSPACE=<...>
+
+    # This copies $LOCAL_WORKSPACE/big_english to /big_english within the PVC.
```

Comment:
Nit: Could you change all `big_english` names to something that more clearly designates a dataset? Perhaps something as simple as `my_dataset` would be good.